tests/contrib/pytest/test_pytest.py

```diff
@@ -94,7 +95,17 @@ def pytest_configure(config):
                         CIVisibility.enable(tracer=self.tracer, config=ddtrace.config.pytest)
                         CIVisibility._instance._itr_meta[ITR_CORRELATION_ID_TAG_NAME] = "pytestitrcorrelationid"
 
-        with override_env(dict(DD_API_KEY="foobar.baz")):
+        if project_dir is None:
+            project_dir = str(self.testdir.tmpdir)
+
+        _test_env = (dict(DD_API_KEY="foobar.baz", CI_PROJECT_DIR=project_dir),)
+        if mock_ci_env:
+            _test_env = _get_pytest_snapshot_gitlab_ci_env_vars(_test_env)
+
+        if block_gitlab_env:
+            _test_env["GITLAB_CI"] = "0"
```

Comment:
Sanity check: `mock_ci_env` is to enable us to run the tests locally but as if in the GitLab CI, and `block_gitlab_env` is to run locally but NOT simulate running it with GitLab. And if both are true, that means we're running it as if in a CI, but not GitLab CI?